src/main/java/org/mule/extension/ftp/internal/ftp/command/ClassicFtpCommand.java

```diff
@@ -79,7 +83,7 @@ protected void doMkDirs(Path directoryPath) {
     String cwd = getCurrentWorkingDirectory();
     Stack<Path> fragments = new Stack<>();
     try {
-      for (int i = directoryPath.getNameCount(); i >= 0; i--) {
+      for (int i = directoryPath.getNameCount(); i > 0; i--) {
```

Comment:
This solves a hidden bug. If you wanted to create a directory as a child of the FTP root, you will end up executing directoryPath.subpath(0, 0) throwing an exception.

There was no test case for that, I've just added one and solved the same for SFTP in https://github.com/mulesoft/mule-ftp-connector/pull/15